ldapdb/models/fields.py

```diff
@@ -228,3 +231,22 @@ def get_prep_lookup(self, lookup_type, value):
         if lookup_type in ('exact',):
             return value
         raise TypeError("DateField has invalid lookup: %s" % lookup_type)
+
+
+class PasswordField(CharField):
+    u""" Field which encodes password like slappasswd """
+    def __init__(self, *args, **kwargs):
+        defaults = {'blank': True,
+                    'db_column': 'userPassword',
+                    'max_length': 128}
+        defaults.update(kwargs)
+        super(fields.CharField, self).__init__(*args, **defaults)
+
+    def get_db_prep_save(self, value, connection):
+        salt = os.urandom(4)
+        h = hashlib.sha1(value)
```

Comment:
It would be good if a query tests which default password hasher is configured in slapd